scripts/can_errgo/error_tui.go

```diff
@@ -0,0 +1,626 @@
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	"golang.org/x/sys/unix"
+)
+
+
+const canFrameSize = 16      // CAN frame size (for a standard CAN frame)
+
+// Converts a 8-byte slice to a uint64, this is for masking purposes.
+func bytesToUint64(b []byte) uint64 {
+    var result uint64
+    for i := 0; i < len(b) && i < 8; i++ {
+        // Reverse the order: least significant byte (rightmost) gets the lowest bits
+        result |= uint64(b[len(b)-1-i]) << (8 * i)
+    }
+    return result
+}
+
+// Default style for the table
+var baseStyle = lipgloss.NewStyle().
+	BorderStyle(lipgloss.NormalBorder()).
+	BorderForeground(lipgloss.Color("240"))
+
+// RowValuesToRow converts a RowValues struct to a table.Row
+func RowValuesToRow(r RowValues) table.Row{
+	return table.Row{r.Error, strconv.Itoa(r.Count), strconv.Itoa(r.Recency)}
+}
+
+// Convert a slice of RowValues to a slice of table.Rows
+func toTableRows(r []RowValues) []table.Row {
+	var rows []table.Row
+	for _, v := range r {
+		// Check if the RowValues is "non-empty" and Active
+		if v.Error != "" && v.Count != 0 && !v.Ignored{
+			rows = append(rows, RowValuesToRow(v))
+		}
+	}
+	return rows
+}
+
+// SortRowValuesBy sorts a slice of RowValues by the specified column, this creates a deep copy of the input slice.
+func SortRowValuesBy(r []RowValues, column int) []RowValues {
+	// Create a deep copy of the input slice
+	copied := make([]RowValues, len(r))
+	copy(copied, r)
+
+	// Sort the copied slice
+	sort.SliceStable(copied, func(i, j int) bool {
+		switch column {
+		case 0:
+			return copied[i].Error < copied[j].Error
+		case 1:
+			return copied[i].Count < copied[j].Count
+		case 2:
+			return copied[i].Recency < copied[j].Recency
+		}
+		return false
+	})
+
+	return copied
+}
+
+// getDescription returns the description of the error, if the error is not found it returns a default message.
+func (m model) getDescription(errorName string) string {
+	for i := range(len(m.rowsValues)){
+		if m.rowsValues[i].Error == errorName{
+			return m.rowsValues[i].Description
+		}
+	}
+	return "No description available"
+}
+
+// findError returns the index of the error in the slice of RowValues, if the error is not found it returns -1.
+func findError(errorName string, rowValues []RowValues) int {
+	cnt := 0
+	for i := range(len(rowValues)){
+		if rowValues[i].Error == ""{
+			cnt += 1
+		} 
+		if rowValues[i].Error == errorName{
+			return i - cnt
+		}
+	}
+	return -1
+}
+
+// resetRowValue resets the count and recency of the error in the slice of RowValues.
+func (m model) resetRowValue(errorName string) {
+	for i := range(len(m.rowsValues)){
+		if m.rowsValues[i].Error == errorName{
+			m.rowsValues[i].Count = 0
+			m.rowsValues[i].Recency = 0
+		}
+	}
+}
+
+// getIgnoredRows returns a slice of table.Rows that are ignored and have an error message.
+func (m model) getIgnoredRows() []table.Row {
+	var rows []table.Row
+	for i := range(len(m.rowsValues)){
```

Comment:
can you iterate over the index and rowValue together so you don't need to constant access `m.rowsValues[i]`?